src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.skia.cs

```diff
@@ -206,21 +201,81 @@ internal override void OnPropertyChanged2(DependencyPropertyChangedEventArgs arg
 					Style = SKPaintStyle.Fill
 				});
 			};
+
+			_inlines.FireDrawingEventsOnEveryRedraw = true;
+			_inlines.RenderSelection = IsTextSelectionEnabled;
+		}
+
+		private void OnKeyDown(KeyRoutedEventArgs args)
+		{
+			switch (args.Key)
+			{
+				case VirtualKey.C when args.KeyboardModifiers.HasFlag(VirtualKeyModifiers.Control):
+					CopySelectionToClipboard();
+					args.Handled = true;
+					break;
+				case VirtualKey.A when args.KeyboardModifiers.HasFlag(VirtualKeyModifiers.Control):
+					SelectAll();
+					args.Handled = true;
+					break;
+			}
+		}
+
+		private void OnDoubleTapped(DoubleTappedRoutedEventArgs e)
+		{
+			if (IsTextSelectionEnabled)
+			{
+				var nullableSpan = Inlines.GetRenderSegmentSpanAt(e.GetPosition(this), false);
+				if (nullableSpan.HasValue)
+				{
+					Selection = new Range(Inlines.GetStartAndEndIndicesForSpan(nullableSpan.Value.span, false));
+				}
+			}
+		}
+
+		// TODO: remove this context menu when TextCommandBarFlyout is implemented
+		private void OnRightTapped(RightTappedRoutedEventArgs e)
+		{
+			e.Handled = true;
+
+			Focus(FocusState.Pointer);
+
+			if (_contextMenu is null)
+			{
+				_contextMenu = new MenuFlyout();
+
+				// TODO: port localized resources from WinUI
```

Comment:
You can now use resources from this file: 

https://github.com/unoplatform/uno/blob/2c29bdfc7819d05359d99373d2e9555aa8ae28a4/src/Uno.UI/Microsoft/UI/Xaml/Controls/WinUIResources/en-us/Resources.resw#L1203C15-L1203C43